src/crimson/os/seastore/object_data_handler.cc

```diff
@@ -297,8 +297,7 @@ overwrite_ops_t prepare_ops_list(
   interval_set<uint64_t> pre_alloc_addr_removed, pre_alloc_addr_remapped;
   if (delta_based_overwrite_max_extent_size) {
     for (auto &r : ops.to_remove) {
-      // TODO: Introduce LBAMapping::is_data_stable() to include EXIST_CLEAN extents
-      if (r->is_stable() && !r->is_zero_reserved()) {
+      if (r->is_data_stable() && !r->is_zero_reserved()) {
```

Comment:
Looks there were 4 places using `is_stable()` in object-data-handler for the remap decisions. They should all be converted to `is_data_stable()` for the same purpose.

Reply:
I think we need to keep the following two` is_stable()`in object-data-handler to give a chance to merge to on-going changes in the transaction. For instance, in multiple_remap test, a total of 10 pins are generated, even though this could be minimized if `is_stable` is used.

```
        is_left_stable(pins.front()->is_stable()),
        is_right_stable(pins.back()->is_stable()) 
```

Reply:
Seem to me it's not that simple.

CC @xxhdx1985126 @zhscn I recall that when `overwrite_plan_t` was introduced, the situation was simple as "delta-based overwrite" didn't exist yet, so that the data extent can only be fresh or stable (or exist-clean).

So the logic related to `overwrite_plan_t::is_left_stable` and `is_right_stable` was quite straitforward:
https://github.com/ceph/ceph/blob/281614e742b9327b7a67ee3992ff113b8347d90d/src/crimson/os/seastore/object_data_handler.cc#L756-L760

That, when an overwrite overlaps a fresh extent (e.g. at left), there is no need to consider splitting because the fresh part haven't submitted to the disk yet. It is intuitive (and efficient) to merge the "overwrite" and fresh part together to form into single larger write activity from the transaction perspective.

If my understanding is correct above, the above decision should still remain correct even after "delta-based overwrite", and `is_left_stable`/`is_right_stable` should become `is_left_fresh`/`is_right_fresh` for the purpose.

The rest of `overwrite_plan_t`, however, I think is no longer accurate because the assumption of write amplification based on `seastore_obj_data_write_amplification` is no longer correct after "delta-based overwrite" is introduced. But I don't think there is any alignment yet about the replacement strategy.

That is to say, in this PR, if I'm corrrect above, how about:
* Mark `is_left_stable`/`is_right_stable` with TODO for `is_left_fresh`/`is_right_fresh`.
* Add notes to the `evaluate_operations()` comment and the description of config `seastore_obj_data_write_amplification` that the strategy might be problematic and needs to be adjusted.

What do you think?

Reply:
Can you check the updated commit to see if I address your comments correctly?